### DISCUSSION TITLE
Clustering with postgres?

### DISCUSSION THREAD

**Discussion by lnd-routing** [lightningnetwork/lnd#5859] [Q&A] [answered]
Is this a planned feature? We are planning to switch to a replicated postgres backend once 0.14 is out, clustering on top would be very useful.

**Comment by guggero** [comment, marked as answer]
With clustering I assume you don't mean to have the _database_ running in cluster mode but `lnd` itself (with active/passive nodes and fast failover)?
That currently isn't a planned feature since (as far as I'm aware) Postgres doesn't have a leader election feature the way `etcd` does. But it should be possible to use `etcd` as the leader election mechanism (for example if you're running on k8s where there already is an `etcd` cluster) but use Postgres as the database backend for `lnd`.
